include/envoy/network/filter.h

```diff
@@ -271,6 +271,28 @@ class ListenerFilterCallbacks {
   virtual void continueFilterChain(bool success) PURE;
 };
 
+/**
+ *  Interface for a listener filter matching with incoming traffic.
+ */
+class ListenerFilterMatcher {
+public:
+  virtual ~ListenerFilterMatcher() = default;
+  virtual bool matches(Network::ListenerFilterCallbacks& cb) const PURE;
+};
+using ListenerFilterMatcherPtr = std::unique_ptr<ListenerFilterMatcher>;
+using ListenerFilterMatcherSharedPtr = std::shared_ptr<ListenerFilterMatcher>;
+
+/**
+ * A mapping from a general listener filter message without `typed_config`.
+ */
+class ListenerFilterConfig {
```

Comment:
Why is this interface needed? Can't the matcher itself just be passed to where it is needed (or null)?

Reply:
For this PR, yes we can. 
However, I feel it's helpful to have common listener filter config besides a config for a particular config.
A reasonable example is that we planned to set timeout for individual listener filter(and avoid adding timeout() for each sniffer listener filter), instead of a listener time out for the entire listener.
WDYT? I can revert to pure matcher if you like. But I do think this ListenerFilterConfig is nice to have. Perhaps renaming to CommonListenerFilterConfig?

Reply:
Please remove until it's actually needed. Thank you.

Reply:
on it